File: android/src/main/java/com/lgzhuo/rct/amap/cluster/Cluster.java
package com.lgzhuo.rct.amap.cluster;

import com.amap.api.maps.model.LatLng;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by yunrui on 2017/4/11.
 */

public class Cluster {
    private LatLng coordinate;
    private List<ClusterPoint> points = new ArrayList<>();

    public Cluster(LatLng coordinate) {
        this.coordinate = coordinate;
    }

    void addPoint(ClusterPoint point){
        this.points.add(point);
    }

    public LatLng getCoordinate() {
        return coordinate;
    }

    public List<ClusterPoint> getPoints() {
        return points;
    }
}

File: ios/RCTAMap/Cluster.m
//
//  Cluster.m
//  Pods
//
//  Created by yunrui on 2017/4/11.
//
//

#import "Cluster.h"

@implementation Cluster
+(instancetype)clusterWithCoordinate:(CLLocationCoordinate2D)coordinate points:(NSArray<ClusterPoint *> *)points{
    Cluster *cluster = [[Cluster alloc]init];
    cluster.coordinate = coordinate;
    cluster.points = points;
    return cluster;
}
@end
